test/s3select_test.cpp

```diff
@@ -217,24 +217,201 @@ TEST(TestS3SElect, arithmetic_operator)
 	ASSERT_EQ( ( (a+b) * (c+d) ).i64() , 21 );
 }
 
-TEST(TestS3SElect, timestamp_function)
+TEST(TestS3selectFunctions, timestamp)
 {
     // TODO: support formats listed here:
     // https://docs.aws.amazon.com/AmazonS3/latest/dev/s3-glacier-select-sql-reference-date.html#s3-glacier-select-sql-reference-to-timestamp
     const std::string timestamp = "2007-02-23:14:33:01";
     // TODO: out_simestamp should be the same as timestamp
     const std::string out_timestamp = "2007-Feb-23 14:33:01";
     const std::string input_query = "select timestamp(\"" + timestamp + "\") from stdin;" ;
-	std::string s3select_res = run_s3select(input_query);
+	  const auto s3select_res = run_s3select(input_query);
     ASSERT_EQ(s3select_res, out_timestamp);
 }
 
-TEST(TestS3SElect, utcnow_function)
+TEST(TestS3selectFunctions, utcnow)
 {
     const boost::posix_time::ptime now(boost::posix_time::second_clock::universal_time());
     const std::string input_query = "select utcnow() from stdin;" ;
-	auto s3select_res = run_s3select(input_query);
+	  auto s3select_res = run_s3select(input_query);
     const boost::posix_time::ptime res_now;
     ASSERT_EQ(s3select_res, boost::posix_time::to_simple_string(now));
 }
 
+TEST(TestS3selectFunctions, add)
+{
+    const std::string input_query = "select add(-5, 0.5) from stdin;" ;
+	  auto s3select_res = run_s3select(input_query);
+    ASSERT_EQ(s3select_res, std::string("-4.500000"));
+}
+
+void generate_csv(std::string& out, size_t size) {
+  // schema is: int, float, string, string
+  std::stringstream ss;
+  for (auto i = 0U; i < size; ++i) {
+    ss << i << "," << i/10.0 << "," << "foo"+std::to_string(i) << "," << std::to_string(i)+"bar" << std::endl;
+  }
+  out = ss.str();
+}
+
+TEST(TestS3selectFunctions, sum)
+{
+    s3select s3select_syntax;
+    const std::string input_query = "select sum(int(_1)), sum(float(_2)) from stdin;";
+    auto status = s3select_syntax.parse_query(input_query.c_str());
+    ASSERT_EQ(status, 0);
+    s3selectEngine::csv_object s3_csv_object(&s3select_syntax);
+    std::string s3select_result;
+    std::string input;
+    size_t size = 128;
+    generate_csv(input, size);
+    status = s3_csv_object.run_s3select_on_object(s3select_result, input.c_str(), size, 
```

Comment:
the run_s3select_on_object accept input size in **bytes**, 
instead of size ==> input.length()

Reply:
ok, got it, will fix!